cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryLogDecider.java

```diff
@@ -50,8 +50,7 @@ public boolean isForReporting(Query query) {
     private boolean executionTimeForLogging(long executionTimeInMilliseconds) {
         logger.trace("Checking if execution time:{} is above threshold: {}", executionTimeInMilliseconds,
                 slowQueryConfiguration.slowQueryThreshold());
-        if (slowQueryConfiguration.logAllQueries()
-                || executionTimeInMilliseconds >= slowQueryConfiguration.slowQueryThreshold()) {
+        if (executionTimeInMilliseconds > slowQueryConfiguration.slowQueryThreshold()) {
```

Comment:
I agree. But my point is that with 0 you won't log all - because of rounding while converting execution time into milliseconds. That's why I'm suggesting handling 0 in a special way.